### PR TITLE
DeepNumpy HuberLoss

### PR DESCRIPTION
**Pull request by goldmermaid**
Base on D2L users' requirement, add huberloss feature for deepnumpy.

### CODE REVIEW

File: python/mxnet/gluon/loss.py [hunk 1]
```diff
@@ -650,9 +650,22 @@ def __init__(self, margin=1, weight=None, batch_axis=0, **kwargs):
...
+
+        if is_np_array():
+            loss = F.np.abs(label - pred)
+            loss = F.np.where(loss > self._rho, loss - 0.5 * self._rho,
+                (0.5 / self._rho) * F.np.square(loss))
+            loss = _apply_weighting(F, loss, self._weight, sample_weight)
+            if F is ndarray:
+                return F.np.mean(loss, axis=tuple(range(1, loss.ndim)))
+            else:
+                return F.npx.batch_flatten(loss).mean(axis=1)
```
Comment (sxjscience) [inline python/mxnet/gluon/loss.py:662]: This won't work for arbitrary batch axis.
